src/App.tsx

```diff
@@ -1,54 +1,123 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [confirmedQuery, setConfirmedQuery] = useState('');
+  const [isVisible, setVisible] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const confirmQuery = useCallback(
+    debounce((value: string) => setConfirmedQuery(value), 1000),
+    [],
+  );
+
+  const onClickHandler
+    = (eventOnClick: React.MouseEvent, person: Person) => {
+      eventOnClick.preventDefault();
+
+      setQuery(person.name);
+      setConfirmedQuery(person.name);
+      setVisible(false);
+      setSelectedPerson(person);
+    };
+
+  const filteredPeople = (people: Person[], queryVal: string) => {
+    const queryNormalized = queryVal.toLocaleLowerCase();
+
+    return people.filter(person => person.name.toLocaleLowerCase()
+      .includes(queryNormalized));
+  };
+
+  const delayOnBlur = () => {
+    setTimeout(() => setVisible(false), 100);
+  };
+
+  const resetHandler = () => {
+    setSelectedPerson(null);
+    setConfirmedQuery('');
+    setQuery('');
+  };
+
+  const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    confirmQuery(event.target.value);
+  };
+
+  const filteredPeopleList = filteredPeople(peopleFromServer, confirmedQuery);
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {selectedPerson ? (
+        <h1 className="title">
+          {`${selectedPerson.name} (${selectedPerson.born} = ${selectedPerson.died})`}
```

Comment:
you can use distruction of Person
const { name, born, died } =  selectedPerson
and use name / born / died in line 58
NOT CRITICAL